app/controllers/navigation/Navigator.scala

```diff
@@ -316,13 +321,19 @@ object Navigator {
     case RepresentativeAgent => representativeAgentClearancePreviousPage
     case IsExs               => isExsClearancePreviousPage
     case Document            => previousDocumentsPreviousPage
+    case DocumentSummary     => previousDocumentsSummaryPreviousPage
   }
 
   val clearanceCacheItemDependent: PartialFunction[DeclarationPage, (ExportsDeclaration, Mode, String) => Call] = {
     case CommodityMeasure   => commodityMeasureClearancePreviousPage
     case PackageInformation => packageInformationClearancePreviousPage
   }
 
+  private def previousDocumentsPreviousPageDefault(cacheModel: ExportsDeclaration, mode: Mode): Call =
+    if (cacheModel.previousDocuments.map(_.documents).getOrElse(Seq.empty).nonEmpty)
```

Comment:
I can't follow this logic.  Does the Navigator have to care about number of existing PreviousDocuments?

Reply:
I'll double check this place

Reply:
The previous documents page to add first document is reused to add next documents too.

So when the previousDocuments are in cache it means that user is adding next document (back link to the previous documents summary page), but when there is no documents in cache it means that it's first document that user is adding (back link to the Nature of transaction page - used only for Standard and Supplementary)

I added additional method which should make code more readable around this logic.